### ISSUE TITLE
digital-twin-registry-0.3.32 → /api/v3.0/lookup/shells?assetIds → Base64 encdinging param doesn't work

### ISSUE THREAD

**Issue by jerzy-wachowiak** [eclipse-tractusx/sldt-digital-twin-registry#395]
# Setup
digital-twin-registry-0.3.32 on AKS
registry.authentication: false 

# Problem
```
curl --location 'https://bpnl00000007txd8.dtr.edc.supplyon.com/semantics/registry/api/v3.0/lookup/shells?assetIds=W3sibmFtZSI6Im1hbnVmYWN0dXJlclBhcnRJZCIsInZhbHVlIjoidXJuOnNhcDpDYXRlbmFSZWZJRCJ9LHsibmFtZSI6ImFzc2V0TGlmZWN5Y2xlUGhhc2UiLCJ2YWx1ZSI6IkFzUGxhbm5lZCJ9XQ' \
--header 'Edc-Bpn: CONSUMER_BPN' \
--header 'Api-Key: xxx'
```
returns
`{"messages":[{"messageType":"Error","text":"Incorrect Base64 encoded value provided as parameter"}]}`

![image](https://github.com/eclipse-tractusx/sldt-digital-twin-registry/assets/29396394/73300b1a-5fbb-4f7a-9931-1c36af0e2ec3)

Example from https://github.com/eclipse-tractusx/sldt-digital-twin-registry/blob/f1f95b927a95f3589991c46b8a16820a964d43dd/backend/src/main/resources/static/aas-registry-openapi.yaml#L712 also fails with the same message

There are no suspicious dtr log entries.

**Comment by jerzy-wachowiak** [comment]
Meanwhile a coworker identified the solution

`assetIds=eyJuYW1lIjoiYXNzZXRMaWZlY3ljbGVQaGFzZSIsInZhbHVlIjoiQXNQbGFubmVkIn0&assetIds=eyJuYW1lIjoibWFudWZhY3R1cmVyUGFydElkIiwidmFsdWUiOiJ1cm46c2FwOkNhdGVuYVJlZklEIn0`

This works and should be documented. The issue can be closed.

**Comment by agg3fe** [comment]
Hi @jerzy-wachowiak, I think the problem is that you have specified externalSubjectId.type as 'GlobalReference', which is not the correct value. The only allowed values are 'ExternalReference' or 'ModelReference'.
If you use this, then you should get the correct result even if you include externalSubjectId in the encoded string.
Here's the link to swagger for your reference https://app.swaggerhub.com/apis/Plattform_i40/AssetAdministrationShellRegistryServiceSpecification/V3.0.1_SSP-001#/Asset%20Administration%20Shell%20Registry%20API/GetAllAssetAdministrationShellDescriptors

**Comment by tunacicek** [comment]
The bug is resolved with edc  version 0.7.X. 
Issue can be closed.
